main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java

```diff
@@ -377,6 +393,116 @@ private void loadFeedInfo(GtfsTableContainer<GtfsFeedInfo> feedTable) {
           localDate.equals(GtfsFeedInfo.DEFAULT_FEED_END_DATE) ? "N/A" : localDate.toString();
       feedInfo.put(FEED_INFO_FEED_END_DATE, info == null ? "N/A" : displayDate);
     }
+
+    loadServiceDateRange(tripContainer, calendarTable, calendarDateTable);
+  }
+
+  /**
+   * Loads the service date range by determining the earliest start date and the latest end date for
+   * all services referenced with a trip\_id in `trips.txt`. It handles three cases: 1. When only
+   * `calendars.txt` is used. 2. When only `calendar\_dates.txt` is used. 3. When both
+   * `calendars.txt` and `calendar\_dates.txt` are used.
+   *
+   * @param tripContainer the container for `trips.txt` data
+   * @param calendarTable the container for `calendars.txt` data
+   * @param calendarDateTable the container for `calendar\_dates.txt` data
+   */
+  public void loadServiceDateRange(
+      GtfsTableContainer<GtfsTrip> tripContainer,
+      GtfsTableContainer<GtfsCalendar> calendarTable,
+      GtfsTableContainer<GtfsCalendarDate> calendarDateTable) {
+    List<GtfsTrip> trips = tripContainer.getEntities();
```

Comment:
Are you sure tripContainer cannot be null?

Reply:
Well, if trips.txt is absent it will give a error notice, but do we have a guarantee that the code here will not be reached in that case? If it is, we'll get an NPE which in general is not good.

Reply:
Ok, it could be empty. I'll add the logic.